docs/ref/forms/format_guidelines.txt

```diff
@@ -0,0 +1,77 @@
+========================
+Form Field Guidelines
+========================
+
+.. _form-field-formats:
+
+Communicating Input Formats
+============================
+
+When using form fields that expect specific formats (like dates, times, or JSON),
+it's important to communicate these requirements to your users. Here are some
+best practices:
+
+Using help_text
+----------------
+
+The ``help_text`` parameter is the recommended way to communicate format requirements:
+
+.. code-block:: python
+
+    date_field = forms.DateField(
+        help_text="Enter date in YYYY-MM-DD format (e.g., 2024-03-21)"
+    )
+
+    time_field = forms.TimeField(
+        help_text="Enter time in HH:MM[:ss[.uuuuuu]] format (e.g., 14:30 or 14:30:00)"
+    )
+
+Common Field Formats
+---------------------
+
+Here are the default formats expected by various Django form fields:
+
+* DateField: :code:`YYYY-MM-DD` (ISO 8601 format)
+* TimeField: :code:`HH:MM[:ss[.uuuuuu]]`
+* DateTimeField: :code:`YYYY-MM-DD HH:MM[:ss[.uuuuuu]][TZ]`
+* DurationField: :code:`DD HH:MM:SS.uuuuuu` or :code:`HH:MM:SS.uuuuuu`
+* JSONField: Valid JSON string
+* UUIDField: :code:`12345678-1234-5678-1234-567812345678`
+* SlugField: URL-friendly string (only letters, numbers, underscores, or hyphens)
+* IPAddressField: :code:`192.168.0.1` (IPv4) or :code:`2001:db8::1` (IPv6)
+
+Best Practices
+---------------
+
+1. Always communicate format requirements upfront through ``help_text`` or labels
+2. Consider using HTML5 input types where supported (e.g., ``type="date"``)
```

Comment:
Shouldn't we be doing this by default by adding `input_type = "date"` to `DateInput` for example?
I think this is related to ticket-36459 (cc @Antoliny0919 )